clients/drcachesim/CMakeLists.txt

```diff
@@ -515,6 +517,9 @@ restore_nonclient_flags(drmemtrace_invariant_checker)
 # get rid of add_win32_flags(target) since that just adds a couple of flags
 # back, and using the ORIG_ values is preferable.
 macro(add_win32_flags target)
+  if (DEBUG)
+    append_property_list(TARGET ${target} COMPILE_DEFINITIONS "DEBUG")
```

Comment:
Part of the flags being clobbered for DR clients and CMake having global or per-file properties.  restore_nonclient_flags() puts them back but there was some reason we couldn't use it for clients and these burst_ tests all have clients in them....this could all be revisited and tried to make cleaner since modern CMake might have better control but that is out of scope for this PR.